src/stirling/source_connectors/socket_tracer/bcc_bpf/openssl_trace.c

```diff
@@ -52,6 +60,29 @@ static __inline void process_openssl_data(struct pt_regs* ctx, uint64_t id,
   process_data(/* vecs */ false, ctx, id, direction, args, bytes_count, /* ssl */ true);
 }
 
+static int get_fd_active_syscall(uint64_t id) {
+  int* fd = ssl_fd_map.lookup(&id);
+  if (fd == NULL) {
+    return kInvalidFD;
+  }
+
+  void** ssl = ssl_userspace_call_map.lookup(&id);
+  if (ssl == NULL) {
+    return kInvalidFD;
+  }
+  // This is a consistency check to verify that the socket fd accessed via the underlying syscall
+  // matches the fd provided to SSL_set_fd. This protects against situations where a non socket
+  // syscall occurs before the SSL_write/SSL_read functions return. We are not sure if this can
+  // happen in practice, however, the socket tracer BPF probes do trigger for generic read/write
+  // syscalls (non socket io).
+  int* expected_fd = openssl_native_bio_map.lookup(ssl);
+  if (expected_fd == NULL || *fd != *expected_fd) {
+    return kInvalidFD;
+  }
+
+  return *fd;
+}
+
```

Comment:
I believe we touched on this in our meeting yesterday, and since we want to move forward with additional testing I'll hold off on responding to this until that testing is complete.

Reply:
`SSL_set_fd` is no longer used since our additional testing from #1161 has confirmed our assumptions. While we are yet to make the final call on that data, I'm going to resolve this since the current direction no longer includes it.